### ISSUE TITLE
Read string typed from keyboard?

### ISSUE THREAD

**Issue by bfreis** [EngoEngine/engo#635]
Is there an easy, built-in way to read a string typed by the user?

I'm thinking something similar to Ebiten's `ebiten.InputChars()`, like in this example: https://hajimehoshi.github.io/ebiten/examples/typewriter.html

Thanks!
Bruno

**Comment by Noofbiz** [comment]
Sadly there isn't anything like that at the moment

**Comment by Noofbiz** [comment]
Sorry lol I accidentally pressed close instead of finishing the comment xD
Anyways, with the action api #626 we were planning on making a UI system that would have text boxes that could do that, and if you wanted to do one on your own that api would be much better than the current one for doing this. At the moment you'd have to query each key and use a switch statement selecting which one. It's kind of a pain. Hopefully soon it will be better though!

**Comment by Noofbiz** [comment]
Fixed in #647 you can now get input characters from engo.TextMessage

**Comment by Inkeliz** [comment]
It's possible to read, but that doesn't stop any other system which listen from the input. It's kind of conflict between systems. 

If you have something like:

```
sys.KeyboardScroller.BindKeyboard(tango.DefaultHorizontalAxis, tango.DefaultVerticalAxis)
```

If you type:

```
Hello World
```

Thre's no issue getting `Hello World` using the message, works fine. However, when you type `W` (from `World`) it will move the camera, since it's a bind key at `KeyboardScroller`.

----

I don't know if already have something to block the actions from the keyboard. I mean, `engo.IgnoreKeyboardInput(true)` (or better name, like `engo.OnlyTyping(true)`).  While that is enabled it will ignore any input from the keyboard. In that case pressing  `W` will only send the message but doesn't make any effect on `KeyboardScroller` or any system which listens that key.
